### PR TITLE
Notice fix

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Notice fix

Before
----------------------------------------

image

After
----------------------------------------

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/Form/AbstractEditPayment.php [hunk 1]
```diff
@@ -669,14 +669,14 @@ protected function assignPaymentInfoBlock() {
...
-  protected function assignContactEmailDetails() {
+  protected function assignContactEmailDetails(): void {
     if ($this->getContactID()) {
       [$displayName] = CRM_Contact_BAO_Contact_Location::getEmailDetails($this->getContactID());
       if (!$displayName) {
         $displayName = civicrm_api3('contact', 'getvalue', ['id' => $this->getContactID(), 'return' => 'display_name']);
       }
-      $this->assign('displayName', $displayName);
     }
+    $this->assign('displayName', $displayName) ?? NULL;
```
Comment (demeritcowboy) [inline CRM/Contribute/Form/AbstractEditPayment.php:679]:
```suggestion
    $this->assign('displayName', $displayName ?? NULL);
```

But noting while this does get rid of the notice, the header text is then missing the name: `Use this form to submit an event registration on behalf of . A LIVE transaction will be submitted using the selected payment processor.` It was missing it before too, but the actual problem is it's trying to display text that doesn't make sense in this situation.

**Comment by eileenmcnaughton** [comment]
thanks @demeritcowboy - I've included your change & tried to care about the other problem you noted

**Comment by demeritcowboy** [comment]
I'm going to merge but I think the tpl line needs to be completely replaced. I would remove the first sentence since it adds no value since right above it already says the name when it's present, and when it's not present is the cause of this notice, and then the second sentence is not correct in other languages (the "live"/"test" is always in english). Elsewhere the approach for that when there's a known list of values is to just hardcode the full sentences in an `{if}` block.

I do think it's important to identify if it's live or test, but since none of the above has been raised as an issue the help text itself seems very unimportant. Maybe it should just have some kind of icon indicating live or test and that's it.

I'll make a lab ticket.

**Comment by demeritcowboy** [comment]
https://lab.civicrm.org/dev/core/-/issues/4809

**Comment by eileenmcnaughton** [comment]
@demeritcowboy yeah - that all makes sense - I tend to think we have too much wordiness but it's hard to see it through the eyes of new users

**Pull request merged**